sdks/java/io/rrio/src/main/java/org/apache/beam/io/requestresponse/RedisClient.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.io.requestresponse;
+
+import static org.apache.beam.sdk.util.Preconditions.checkStateNotNull;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.joda.time.Duration;
+import redis.clients.jedis.JedisPooled;
+import redis.clients.jedis.exceptions.JedisException;
+
+/**
+ * {@link RedisClient} is a convenience class that supports operations needed for caching by various
+ * transforms in this package. It implements the {@link SetupTeardown} interface for ease-of-use
+ * within a {@link DoFn} context. Unlike the underlying {@link JedisPooled} client, {@link
+ * RedisClient} is {@link java.io.Serializable}.
+ */
+class RedisClient implements SetupTeardown {
```

Comment:
Its a little strange that none of these methods are public or private

Reply:
They are needed as package private. Private would prevent their use by other classes in the `org.apache.beam.io.requestresponse` package. Public would expose it outside `org.apache.beam.io.requestresponse` and there isn't _yet_ a reason to do this. This class will be used by `CacheRead` and `CacheWrite`, for example, classes that are package private. Could you tell me why you think there might be value in making this public?

Reply:
@johnjcasey I will go ahead and resolve this and thank you again for reviewing. Please re-open if you feel strongly about something I may have missed regarding exposing these methods via the public modifier.